### DISCUSSION TITLE
Embed rbfx by add_subdirectory， What exactly happened？

### DISCUSSION THREAD

**Discussion by luciddream-tsin** [rokups/rbfx#461] [Q&A] [answered]
If I put my client project in a separate folder, it can't find the engine file, but if I follow the tutorial on the wiki, it was ok.
I can't find where the include operation is, what is the key point that makes these differences? thanks.

![截图 2022-08-25 17-47-32](https://user-images.githubusercontent.com/92364028/186633216-235936dc-440a-483d-a654-8ea021807530.png)

**Comment by eugeneko** [comment, marked as answer]
> I can't find where the include operation is, what is the key point that makes these differences? thanks.

When you link the Urho3D library to your executable, CMake should initialize all the dependencies.
Make sure you do `target_link_libraries`.

We don't (yet) have official sample project, but I test engine integration on this project, you can check what I do in CMakeFiles:
https://github.com/eugeneko/Snake4D

**Reply by luciddream-tsin** [reply]
That project is exactly what I want， thanks!
